### ISSUE TITLE
EventData schema: Expose SystemProperties/Properties

### ISSUE THREAD

**Issue by SreeramGarlapati** [Azure/azure-event-hubs-spark#250]
Exposing `SystemProperties` will help us deliver amqp message annotations to EventDataSchema. `IOT-client` relies on this feature to deliver `device-id` to their customers.

```
def eventHubsSchema: StructType = {
    StructType(
      Seq(
        StructField("body", BinaryType),
        StructField("systemProperties", MapType(StringType, AnyDataType())),
     ...... 
      ))
  }
```

As I tried to highlight - 
1) we should expose this as a key value pair of string, object and 
2) remove the 3 standard sysproperties (offset, ...) to de-duplicate the payload size

**Comment by sabeegrewal** [comment]
I think I'm going to skip this for now - it seems system properties always has `offset`, `enqueued time`, `sequence number`, and `publisher`. It's much more legible to just unpack that for the user. 

The properties bag can change depending on the particular event, so it makes sense to display that as a map (#251).

**Comment by sabeegrewal** [comment]
Re-opening. Looks like SystemProperties can be more than just those four fields.

**Comment by ogidogi** [comment]
Any plans when 2.3.2 (with SystemProperties/Properties) will be released?

**Comment by sabeegrewal** [comment]
@ogidogi after talking with some Spark project members, it seems to best way to go was to serialize the map values as json strings. I just opened #328 with the changes. Do you have time to check to see if this meets your requirements? I should be able to get this included in `2.3.2`.

**Comment by ogidogi** [comment]
Hi, sabeegrewal it's more then enough. Will build a new version tomorrow and try it.

**Comment by sabeegrewal** [comment]
@ogidogi - sounds good! I'll try to have all the open PRs merged by tomorrow. 👍

**Comment by ogidogi** [comment]
Check version which has been built form this commit on Master:
Commit: 4cf407258871fc6639957eced9323cd0f951748b [4cf4072]
Parents: 2c79b5f05f
Author: Sabee Grewal <grewalsabee@gmail.com>
Date: Friday, June 8, 2018 21:26:22
Committer: GitHub
Javadocs (#337)

Executed everything on Spark 2.3.1 - all works fine, have not find any issues yet. Everything works as expected.
Thank a lot.

When we can expect azure-eventhubs-spark_2.11 v2.3.2 will be in maven?

**Comment by sabeegrewal** [comment]
Sounds good! I'm in the middle of a parallelization effort right now so that the connector scales better. That's turning out to be more work than I expected since I'm looking into service + API improvements as well. I'm likely going to merge the open PR (which only has very minor improvements) and release a 2.3.2. Then I'll release the fully improved version in 2.3.3 once the performance improvements are fully complete. 

So, long story short, I'm hoping to release 2.3.2 late next week!
